src/v/cluster/health_monitor_backend.cc

```diff
@@ -151,16 +159,62 @@ std::optional<node_health_report> health_monitor_backend::build_node_report(
     return report;
 }
 
+void health_monitor_backend::abortable_refresh_request::abort() {
+    if (finished) {
+        return;
+    }
+    finished = true;
+    done.set_value(errc::timeout);
+}
+
+health_monitor_backend::abortable_refresh_request::abortable_refresh_request(
+  model::node_id leader_id, ss::gate::holder holder, ss::semaphore_units<> u)
+  : leader_id(leader_id)
+  , holder(std::move(holder))
+  , units(std::move(u)) {}
+
+void health_monitor_backend::abortable_refresh_request::set_fiber(
```

Comment:
Since an abortable_refresh_request with no fiber doesn't make sense, maybe the fiber should be set as part of the constructor instead.

Reply:
I've end up with:
```c++
ss::future<std::error_code> abortable_await(ss::future<std::error_code>);
```